formcentral/db.go

```diff
@@ -54,3 +64,27 @@ func getTranslationForms(pool *pgxpool.Pool, surveyid string) (*trans.FormJson,
 
 	return src, dest, err
 }
+
+func getSurveysByPageID(pool *pgxpool.Pool, pageid string, code string, created string) ([]Survey, error) {
+   query := `
+      SELECT id, userid, form_json, shortcode, translation_conf, created
+      FROM surveys
+      WHERE userid=(SELECT userid FROM credentials WHERE facebook_page_id=$1 LIMIT 1)
+      AND shortcode=$2
+      AND created<=$3
+      ORDER BY created DESC
+   `
+   rows, err := pool.Query(context.Background(), query, pageid, code, created)
+
+   surveys := []Survey{}
+   for rows.Next() {
+      s := Survey{}
+      err := rows.Scan(&s.ID, &s.Userid, &s.Form_json, &s.Shortcode, &s.Translation_conf, &s.Created)
+      if err != nil {
+         fmt.Println(err)
```

Comment:
Should return the error if some row can't be scanned, as that shouldn't happen!

Reply:
_used `defer`, see https://github.com/vlab-research/fly/pull/24/commits/5ed3ca59187e754132911655daff0ffaaf5fee59_